aws/sqs_event_manager/sqs_event_manager/app.py

```diff
@@ -0,0 +1,133 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with mash.  If not, see <http://www.gnu.org/licenses/>
+#
+"""
+Lambda to handle SQS messages from from
+AWS Metering/Entitlement Marketplace API's
+"""
+import json
+import logging
+
+from sqs_event_manager.queue import delete_message
+from sqs_event_manager.message import AWSSNSMessage
+from resolve_customer.entitlements import AWSCustomerEntitlement
+
+logger = logging.getLogger('sqs_event_manager')
+logger.setLevel('INFO')
+
+
+def lambda_handler(event, context):
+    """
+    Expects event type matching the SNS topic
+
+    {
+        'Records': [
+            {
+                'message_id': 'str',
+                'receipt_handle': 'str',
+                'body': {
+                    'Type' : 'str',
+                    'MessageId : 'str',
+                    'TopicArn' : 'str',
+                    'Message' : {
+                        'action': 'str',
+                        'customer-identifier': 'str',
+                        'product-code': 'str',
+                    },
+                    'Timestamp' : 'str',
+                    'SignatureVersion' : 'str',
+                    'Signature' : 'str',
+                    'SigningCertURL' : 'str',
+                    'UnsubscribeURL' : 'str'
+                },
+                'attributes': {
+                    'ApproximateReceiveCount': 'str',
+                    'SentTimestamp': 'str',
+                    'SenderId': 'str',
+                    'ApproximateFirstReceiveTimestamp': 'str'
+                },
+                'messageAttributes': {},
+                'md5OfBody': 'str',
+                'eventSource': 'aws:sqs',
+                'eventSourceARN': 'arn:aws:sqs:us-east-1:111122223333:my-queue',
+                'awsRegion': 'us-east-1'
+            }
+        ]
+    }
+    """
+    batch_item_failures = []
+    sqs_batch_response = {}
+
+    try:
+        for message in event['Records']:
+            process_message(message, batch_item_failures)
+    except Exception as error:
+        return json.dumps(
+            error_response(500, f'{type(error).__name__}: {error}')
+        )
+
+    sqs_batch_response['batchItemFailures'] = batch_item_failures
+    return json.dumps(sqs_batch_response)
+
+
+def process_message(record: dict, batch_item_failures: dict):
+    """
+    Handle message received from SQS queue
+
+    Ensure the message is proper format. Get the customer entitlements and
+    send the information to the SCC service.
+    """
+    try:
+        message = AWSSNSMessage(record)
+
+        if not message.action:
+            logger.info(f'Received an unknown message: {json.dumps(record)}')
+        elif message.action == 'entitlement-updated':
+            # Notify SCC of customer entitlement change for the given product
+            entitlements = AWSCustomerEntitlement(
```

Comment:
@schaefi @rjschwei Do we want to import the resolve customer SDK or make a request to the service?

Reply:
@schaefi if we go with the import how do we want to handle the dependency resolution?

Reply:
Add the following to your pyproject.toml

```patch
diff --git a/aws/sqs_event_manager/pyproject.toml b/aws/sqs_event_manager/pyproject.toml
index 3057740..3247951 100644
--- a/aws/sqs_event_manager/pyproject.toml
+++ b/aws/sqs_event_manager/pyproject.toml
@@ -41,6 +41,7 @@ classifiers = [
 python = "^3.11"
 requests = ">=2.25.0"
 setuptools = ">=50"
+resolve_customer = { git = "https://github.com/SUSE-Enceladus/suse-saas-tools.git", subdirectory = "aws/resolve_customer", branch = "main" }
 
 [tool.poetry.scripts]
 # call scripts might be not needed, but just to show how
```

and in your `package/python-sqs_event_manager-spec-template` add

```
BuildRequires:  %{python_module resolve_customer}
Requires: %{python_module resolve_customer}
```

Reply:
call `poetry lock` to update dependencies

Reply:
Thanks, I'm not too familiar with handling those style dependencies in poetry